ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -3,15 +3,10 @@
 //! This is a parser for the on-disk (in the ELF binary) IR format used to express the
 //! ahead-of-time compiled interpreter.
 //!
-//! The `Display` implementations convert the in-memory data structures into a human-readable
-//! textual format.
+//! Once decoded, the IR is immutable. Any mutations after the fact are UB.
```

Comment:
> Any mutations after the fact are UB.

Why do we allow mutations if it's UB? Why don't we hide anything which enables mutation?

Reply:
Once we've passed it outside of this module it's opaque and immutable.

It's more a not for developers working within this crate, who will have free reign. Perhaps the module doc isn't the right place for my note?

Reply:
>Perhaps the module doc isn't the right place for my note?

Agreed: it shouldn't be in the externally facing docs.

Reply:
Fixed in 20b53e05b6db8e1a775d99eb084c9be36505016a.